### ISSUE TITLE
ci: Prefer certain builds

### ISSUE THREAD

**Issue by jalil-salame** [wez/wezterm#3360]
## Problem

Current CI is very expansive, so runners may pick up builds in an inconvenient order.

## Proposed solution

Use [`workflow_run`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run):

1. Select Linux Targets to be priorized.
2. Add yaml to non priorized Linux Targets, referencing priorized targets:

   ```yaml
   on:
     workflow_run:
       workflows: ["ubuntu_build", "centos_build"]
       type: ["in_progress"]
   ```

This will make it so that those Linux targets wait until any of the priorized targets are running before starting.

It might still clog the CI, but its worth a try.

## Discussion questions

- Which distros should be priorized?
  - `opensuse_tumbleweed`, `alpine` and `ubuntu` seem to be the fastest workflows (by ~3-4 minutes)
- Should we instead use `on_completed`? This might ensure that the other CI workers are running.
- `workflow_run` doesn't support `path` triggers, only branch triggers, my suggestion is to add a `push.paths = .github/workflows/this_workflow.yaml` trigger, this will clog the CI if the workflows are updated, but it will ensure they are triggered if the priorized workflows aren't triggered.

**Comment by wez** [comment]
If we can prioritize the latest ubuntu, macos and windows builds that would be good. Using completion (regardless of success/failure) as the trigger for the remainder sounds fine to me.  The suse and alpine builds are occasionally flaky and I think they have a much smaller user base, so I consider them lower priority.

I'm fine having a push.paths = workflow.yml trigger for those; in theory, sweeping updates to all the workflows are rare (except for all the work you're doing right now!) so it's ok and expected for them to trigger a bunch of work all at once when those do change.
